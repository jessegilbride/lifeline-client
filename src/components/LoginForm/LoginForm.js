import React, { Component } from 'react';
import AuthApiService from '../../services/auth-api-service';
// import TimelineApiService from '../../services/timeline-api-service';

export default class LoginForm extends Component {
  static defaultProps = {
    onLoginSuccess: () => {}
  }

  state = { error: null }

  handleSubmitJwtAuth = event => {
    event.preventDefault()
    this.setState({ error: null })
    const { user_name, password } = event.target

    AuthApiService.postLogin({
      user_name: user_name.value,
      password: password.value,
    })
    /* .then(res => {
      window.localStorage.setItem(`username`, user_name.value)

      TimelineApiService.getFullNameByUsername(user_name.value)
      .then(res => {
        console.log(res)
        return res
      })
      .then(fullname => window.localStorage.setItem(`fullname`, fullname))
    }) */
    .then(res => {
      user_name.value = ''
      password.value = ''
      this.props.onLoginSuccess()
    })
    .catch(res => {
      this.setState({ error: res.error })
    })
  }

  render() {
    const { error } = this.state
    return (
      <form
        className='LoginForm'
        onSubmit={this.handleSubmitJwtAuth}
      >
        <div role='alert'>
          {error && <p className='error-message'>{error}</p>}
        </div>
        <div className='user_name form-section'>
          <label htmlFor='LoginForm__user_name'>
            User name
          </label>
          <input
            required
            name='user_name'
            defaultValue='demoUser'
            id='LoginForm__user_name'>
          </input>
        </div>
        <div className='password form-section'>
          <label htmlFor='LoginForm__password'>
            Password
          </label>
          <input
            required
            name='password'
            type='password'
            defaultValue='Hello123'
            id='LoginForm__password'>
          </input>
        </div>
        <button type='submit'>
          Login
        </button>
      </form>
    )
  }
}
