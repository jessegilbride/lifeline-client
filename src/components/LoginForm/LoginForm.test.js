import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from "react-router-dom";
import renderer from 'react-test-renderer';
import LoginForm from './LoginForm';

describe('LoginForm component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
    <MemoryRouter><LoginForm /></MemoryRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
      <MemoryRouter><LoginForm name="LoginForm" /></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

});
