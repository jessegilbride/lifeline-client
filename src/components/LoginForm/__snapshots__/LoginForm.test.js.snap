// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders the UI as expected 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="user_name form-section"
  >
    <label
      htmlFor="LoginForm__user_name"
    >
      User name
    </label>
    <input
      defaultValue="demoUser"
      id="LoginForm__user_name"
      name="user_name"
      required={true}
    />
  </div>
  <div
    className="password form-section"
  >
    <label
      htmlFor="LoginForm__password"
    >
      Password
    </label>
    <input
      defaultValue="Hello123"
      id="LoginForm__password"
      name="password"
      required={true}
      type="password"
    />
  </div>
  <button
    type="submit"
  >
    Login
  </button>
</form>
`;
